src/python.cc

```diff
@@ -393,16 +387,22 @@ ModelInstanceState::~ModelInstanceState() {
   int status;
   kill(interpreter_pid_, SIGTERM);
   waitpid(interpreter_pid_, &status, 0);
-  unlink(domain_socket_.substr(7).c_str());
 
-  // FIXME currently GRPC client uses an async thread for cleaning up and
-  // shutting down the connection, however, as reported in
-  // https://github.com/grpc/grpc/issues/22479 the clean up thread may continue
-  // to live after resources have been deallocated an cause a segfault. This is
-  // a workaround to do a blocking shutdown of the GRPC client
-  grpc_shutdown_blocking();
+  // Check if the domain socket has been set.
+  if (domain_socket_.length() >= 7) {
```

Comment:
The domain socker it is either never set i.e. "" (if the temp dir cannot be created) or set to `unix://tmp_dir_name/unix.socket` Shouldn't this simply check if it is empty?

Reply:
Sure, I will change it to a simple empty check.